tests/common/include/tests/common/tools_helix_intersectors.inl

```diff
@@ -0,0 +1,138 @@
+
+    // Track defined on origin point
+    const free_track_parameters<transform3> free_trk(
+        {0.f, 0.f, 0.f}, 0.f, {0.1f * unit<scalar>::GeV, 0.f, 0.f}, -1.f);
+
+    // Magnetic field
+    const vector3 B{0.f, 0.f, 1.f * unit<scalar>::T};
+
+    const detail::helix<transform3> hlx(free_trk, &B);
+
+    const scalar path = 10.f * unit<scalar>::cm;
+
+    // Transform translation vector
+    const auto trl = hlx(path);
+
+    // Surface normal vector
+    const auto w = hlx.dir(path);
+
+    // z axis
+    const vector3 z_axis{0.f, 0.f, 1.f};
+
+    // Vector on the surface
+    const auto v = vector::cross(z_axis, w);
+
+    // Transform matrix
+    const transform3 trf(trl, w, v);
+
+    // Rectangle surface
+    const detray::mask<detray::rectangle2D<>> rectangle{
+        0u, 10.f * unit<scalar>::cm, 10.f * unit<scalar>::cm};
+
+    const helix_plane_intersector<transform3> hpi;
+
+    // Get the intersection on the next surface
+    const auto is = hpi(hlx, rectangle, trf)[0];
+
+    // Check the values
+    EXPECT_NEAR(is.path, path, tolerance);
+    EXPECT_NEAR(is.p2[0], 0.f, tolerance);
+    EXPECT_NEAR(is.p2[1], 0.f, tolerance);
+    EXPECT_NEAR(is.p3[0], trl[0], tolerance);
+    EXPECT_NEAR(is.p3[1], trl[1], tolerance);
+    EXPECT_NEAR(is.p3[2], trl[2], tolerance);
+}
+
+TEST(tools, helix_cylinder_intersector) {
+
+    // Track defined on origin point
+    const free_track_parameters<transform3> free_trk(
+        {0.f, 0.f, 0.f}, 0.f, {0.1f * unit<scalar>::GeV, 0.f, 0.f}, -1.f);
+
+    // Magnetic field
+    const vector3 B{0.f, 0.f, 1.f * unit<scalar>::T};
+
+    const detail::helix<transform3> hlx(free_trk, &B);
+
+    const scalar path = 10.f * unit<scalar>::cm;
+
+    // Transform translation vector
+    const auto trl = hlx(path);
+
+    // Vector on the surface
+    const auto v = hlx.dir(path);
+
+    // z axis
+    const vector3 z_axis{0.f, 0.f, 1.f};
+
+    // Transform matrix
+    const transform3 trf(trl, z_axis, v);
+
+    const scalar c_rad = 5.f * unit<scalar>::cm;
+
+    // Cylinder surface (5 cm radius)
+    const detray::mask<detray::cylinder2D<>> cylinder{
+        0u, c_rad, 10.f * unit<scalar>::cm, 10.f * unit<scalar>::cm};
+
+    const helix_cylinder_intersector<transform3> hci;
+
+    // Get the intersection on the next surface
+    const auto is = hci(hlx, cylinder, trf);
+    const auto& is_front = is[0];
+    const auto pos_front = hlx.pos(path - c_rad);
+    const auto loc_front = pos_front - trl;
+    const auto phi_front =
+        std::acos(vector::dot(v, loc_front) /
+                  (getter::norm(v) * getter::norm(loc_front)));
+
+    EXPECT_NEAR(is_front.path, path - c_rad, tolerance);
+    EXPECT_NEAR(is_front.p2[0], c_rad * phi_front, tolerance);
+    EXPECT_NEAR(is_front.p2[1], 0.f, tolerance);
+    EXPECT_NEAR(is_front.p3[0], pos_front[0], tolerance);
+    EXPECT_NEAR(is_front.p3[1], pos_front[1], tolerance);
+    EXPECT_NEAR(is_front.p3[2], pos_front[2], tolerance);
+
```

Comment:
@niermann999 The difference with the truth value seems not small for cylinder intersector - is it meant to be like this? 
```
/home/beomki/projects/detray/detray/tests/common/include/tests/common/tools_helix_intersectors.inl:117: Failure
The difference between is_front.path and path - c_rad is 0.0469207763671875, which exceeds tolerance, where
is_front.path evaluates to 49.953079223632812,
path - c_rad evaluates to 50, and
tolerance evaluates to 9.9999997473787516e-05.
/home/beomki/projects/detray/detray/tests/common/include/tests/common/tools_helix_intersectors.inl:118: Failure
The difference between is_front.p2[0] and c_rad * phi_front is 0.003570556640625, which exceeds tolerance, where
is_front.p2[0] evaluates to 153.32867431640625,
c_rad * phi_front evaluates to 153.33224487304688, and
tolerance evaluates to 9.9999997473787516e-05.
/home/beomki/projects/detray/detray/tests/common/include/tests/common/tools_helix_intersectors.inl:120: Failure
The difference between is_front.p3[0] and pos_front[0] is 0.04639434814453125, which exceeds tolerance, where
is_front.p3[0] evaluates to 49.766578674316406,
pos_front[0] evaluates to 49.812973022460938, and
tolerance evaluates to 9.9999997473787516e-05.
/home/beomki/projects/detray/detray/tests/common/include/tests/common/tools_helix_intersectors.inl:121: Failure
The difference between is_front.p3[1] and pos_front[1] is 0.0069985389709472656, which exceeds tolerance, where
is_front.p3[1] evaluates to 3.7333962917327881,
pos_front[1] evaluates to 3.7403948307037354, and
tolerance evaluates to 9.9999997473787516e-05.
```

Reply:
Ah sorry I should not use `c_rad` here

Reply:
Hm, come to think of it, I think I added some checks in #411, too... https://github.com/acts-project/detray/pull/411/files#diff-794d995a8b19d1070c691fcb6473c827e09f268dd83c8247b4ff03704d7edcef

Reply:
The helix intersector needed adjustments to be able to find both solutions

Reply:
So helix intersector is supposed to have two solutions but not implemented yet?

Reply:
Yes, all cylinder intersectors are supposed to have two solutions, right? #411 is only stuck on some performance issues...

Reply:
I need to adjust the path though and test it

Reply:
Let's keep this test anyway, I like it